packages/dapp/components/redeem/lib/use-prices.ts

```diff
@@ -17,15 +16,13 @@ const usePrices = (): [BigNumber | null, BigNumber | null, () => Promise<void>]
         return;
       }
 
-      if(protocolContracts.managerFacet && protocolContracts.twapOracleDollar3poolFacet) {
-        const dollarTokenAddress = await protocolContracts.managerFacet.dollarTokenAddress();
-        const newTwapPrice = await protocolContracts.twapOracleDollar3poolFacet.consult(dollarTokenAddress);
-        const dollar3poolMarket = await protocolContracts.managerFacet.stableSwapMetaPoolAddress();
-        const dollarMetapool = getIMetaPoolContract(dollar3poolMarket, provider)
-        const newSpotPrice = await dollarMetapool["get_dy(int128,int128,uint256)"](0, 1, utils.parseEther("1"));
-        setTwapPrice(newTwapPrice);
-        setSpotPrice(newSpotPrice);
-      }
+      const contracts = await protocolContracts;
+
+      const dollarTokenAddress = await contracts.managerFacet!.dollarTokenAddress();
```

Comment:
operator `!` again

Reply:
oh.. those are fixed in [PR809](https://github.com/ubiquity/ubiquity-dollar/pull/809)

Reply:
it is for the whole codebase, not just the PR